inc/WC_JumiaPay_Purchase.php

```diff
@@ -125,7 +125,16 @@ public function generateData() {
                 "mobilePhoneNumber"=> wc_sanitize_phone_number($this->order->get_billing_phone())
             ]
         ];
+        
+        return $this->filterNotNull($data);
+    }
 
-        return array_filter( $data, function( $v ) { return !( is_null( $v) or '' === $v ); } );
+    private function filterNotNull($data) {
+      $data = array_map(function($item) {
+          return is_array($item) ? $this->filterNotNull($item) : $item;
+      }, $data);
+      return array_filter($data, function($item) {
```

Comment:
as a clean code, assign the return expression to a variable and return that variable

Reply:
actually it will return a sonarlint error if I do so